cradle-cassandra/src/main/java/com/exactpro/cradle/cassandra/dao/testevents/PageScopesIteratorProvider.java

```diff
@@ -0,0 +1,79 @@
+package com.exactpro.cradle.cassandra.dao.testevents;
+
+import com.datastax.oss.driver.api.core.cql.BoundStatementBuilder;
+import com.exactpro.cradle.BookInfo;
+import com.exactpro.cradle.cassandra.dao.CassandraOperators;
+import com.exactpro.cradle.cassandra.dao.testevents.converters.PageScopeEntityConverter;
+import com.exactpro.cradle.cassandra.iterators.ConvertingPagedIterator;
+import com.exactpro.cradle.cassandra.resultset.IteratorProvider;
+import com.exactpro.cradle.cassandra.retries.SelectQueryExecutor;
+import com.exactpro.cradle.counters.Interval;
+
+import java.time.Instant;
+import java.util.*;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+public class PageScopesIteratorProvider extends IteratorProvider<String> {
+
+
+    private final CassandraOperators operators;
+    private final ExecutorService composingService;
+    private final SelectQueryExecutor selectQueryExecutor;
+    private final BookInfo book;
+    private final Queue<String> pages;
+    private final Function<BoundStatementBuilder, BoundStatementBuilder> readAttrs;
+
+    public PageScopesIteratorProvider(String requestInfo,
+                                      CassandraOperators operators,
+                                      BookInfo book,
+                                      Interval interval,
+                                      ExecutorService composingService,
+                                      SelectQueryExecutor selectQueryExecutor,
+                                      Function<BoundStatementBuilder, BoundStatementBuilder> readAttrs) {
+        super(requestInfo);
+        this.operators = operators;
+        this.book = book;
+        this.pages = getPagesInInterval(book, interval);
+        this.composingService = composingService;
+        this.selectQueryExecutor = selectQueryExecutor;
+        this.readAttrs = readAttrs;
+    }
+
+    private Queue<String> getPagesInInterval(BookInfo book, Interval interval) {
+        Instant start = interval.getStart();
+        Instant end = interval.getEnd();
+
+        return book.getPages()
+                .stream()
+                .filter(page -> (page.getStarted().isAfter(start) && page.getStarted().isBefore(end))
+                        || (page.getEnded().isAfter(start) && page.getEnded().isBefore(end)))
+                .map(page -> page.getId().getName())
+                .collect(Collectors.toCollection(LinkedList::new));
```

Comment:
```suggestion
                .collect(Collectors.toCollection(ArrayList::new));
```

Reply:
ArrayList can't be used for queue

Reply:
@vdonadze Maybe we can lazily iterate over the pages instead of collecting them in a queue?
Also, do we really hold all pages in memory? Don't we need to make a request to Cassandra to get all data?

Reply:
you were right about loading pages from Cassandra, I corrected that at the moment. I will check about lazily iterate over the pages

Reply:
@vdonadze I think there should be an example how to do that in Message iterators